### PR TITLE
Array data type has been implemented

### PR DESCRIPTION
**Pull request by baybatu**
Array data type has been implemented for JS interpreter. Also input and output values must be valid json from now on. For example if we want to input integer type, then we must enter without quote or anything like `31`, `55`...

If we want to enter string data type value, then we must enter our string with quote surrounded like other programming languages such as `"merhaba"`, `"naber"`...

If we want to enter array data type value, then we must enter our array value like json array. such as `[1, 2, 3, "naber", "expercise"]`.

Also we have to update current input and output values in DB through an SQL script.

### CODE REVIEW

File: src/main/java/com/expercise/domain/challenge/Challenge.java [hunk 1]
```diff
@@ -240,17 +236,17 @@ public void setSignatures(Map<Lingo, String> signatures) {
...
 
-    public Object getOutputValueFor(TestCase testCase) {
-        return outputType.convert(testCase.getOutput());
-    }
-
     public boolean isCodeKata() {
         return getChallengeType() == ChallengeType.CODE_KATA;
     }
 
+    public boolean hasOneInput() {
```
Comment (ufuk) [inline src/main/java/com/expercise/domain/challenge/Challenge.java]: hasOnlyOneInput() might be a better name

File: src/main/java/com/expercise/interpreter/typechecker/javascript/ListJavaTypeWrapper.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+public class ListJavaTypeWrapper extends BasicJavaTypeWrapper {
+
+    public ListJavaTypeWrapper(Class<?> javaClass) {
+        super(javaClass);
+    }
+
+    @Override
+    boolean valid(Object value) {
+        return Collection.class.isInstance(value) ||
+                (value instanceof ScriptObjectMirror && List.class.isInstance(((ScriptObjectMirror) value).values()));
```
Comment (ufuk) [inline src/main/java/com/expercise/interpreter/typechecker/javascript/ListJavaTypeWrapper.java]: We should eliminate Nashorn dependency.

Comment (baybatu) [inline src/main/java/com/expercise/interpreter/typechecker/javascript/ListJavaTypeWrapper.java]: BasicJavaTypeWrapper and ListJavaTypeWrapper classes are JavaScript specific type wrappers. Therefore i put them all in 'javascript' package.

Comment (ufuk) [inline src/main/java/com/expercise/interpreter/typechecker/javascript/ListJavaTypeWrapper.java]: I did not notice the package name. Thank you.

**Pull request merged**